glfw-test.rkt

```diff
@@ -1,58 +1,60 @@
-
-  (test-case "Opening window in windowed mode succeeds"
-             (glfwOpenWindow 400 300
-                             8 8 8
-                             0 0 0
-                             GLFW_WINDOW)
-             (glfwCloseWindow))
-
-  (test-case "Opening window in fullscreen mode succeeds"
-             (glfwOpenWindow 640 480
-                             8 8 8
-                             0 0 0
-                             GLFW_FULLSCREEN)
-             (glfwCloseWindow)))
-
-(define-test-suite glfw-window-functions
-  #:before (lambda ()
-             (glfwInit)
-             (glfwOpenWindow 400 300
-                             8 8 8
-                             0 0 0
-                             GLFW_WINDOW))
-  #:after (lambda ()
-            (glfwCloseWindow)
-            (glfwTerminate))
-
-  (test-case "Setting and getting window size succeeds"
-             (glfwSetWindowSize 350 250)
-             (let-values ([(x y) (glfwGetWindowSize)])
-               (check-eq? x 350)
-               (check-eq? y 250)))
-
-  (test-case "Setting window position succeeds"
-             (glfwSetWindowPos 10 10))
-
-  (test-case "Setting window title succeeds"
-             (glfwSetWindowTitle "Test title"))
-
-  (test-case "Window iconify and restore succeeds"
-             (glfwIconifyWindow)
-             (glfwRestoreWindow)))
-
-(run-tests glfw-init)
-(run-tests glfw-window-functions)
+
+(module+ test
+  (require "glfw.rkt"
+           disposable
+           fixture
+           racket/function
+           rackunit
+           rackunit/text-ui)
+
+  (define (check-not-= x y) (check-false (= x y)))
+
+  (define-test-suite initialization
+    #:before glfwInit
+    #:after glfwTerminate
+
+    (test-case "Version query succeeds with non-zero values"
+               (let-values ([(major minor rev) (glfwGetVersion)])
+                 (check-not-= major 0)
+                 (check-not-= minor 0)
+                 (check-not-= rev 0)))
+
+    (test-case "Opening window in windowed mode succeeds"
+               (let ([window (glfwCreateWindow 400 300 "Main window" #f #f)])
+                 (glfwDestroyWindow window)))
+
+    (test-case "Opening window in fullscreen mode succeeds"
+               (let ([window (glfwCreateWindow 400 300 "Main window" (glfwGetPrimaryMonitor) #f)])
+                 (glfwDestroyWindow window))))
+
+  (define (create-window)
+    (glfwInit)
+    (glfwCreateWindow 400 300 "Main window" #f #f))
+  (define (destroy-window window)
+    (glfwDestroyWindow window)
+    (glfwTerminate))
+
+  (define-fixture window (disposable create-window destroy-window))
+
+  (test-case/fixture "window-functions"
+    #:fixture window
+    (test-case "Setting and getting window size succeeds"
+               (glfwSetWindowSize (current-window) 350 250)
+               (sleep 1) ; There is a slight delay before the size is actually set
+               (let-values ([(x y) (glfwGetWindowSize (current-window))])
```

Comment:
Oh hey, there's a [rackunit issue](https://github.com/racket/rackunit/issues/71) to make this sort of multiple-values testing easier. Neat to see another use case in the wild :)